File: src/pooh3/mobi/app/a01_clock/model/ShortHand.java
package pooh3.mobi.app.a01_clock.model;

import static pooh3.mobi.Preconditions.checkArgument;
import static pooh3.mobi.Preconditions.checkNotNull;

public class ShortHand {
    public final int hour;
    public ShortHand(int hour) {

        checkArgument(hour >= 0,
                "hour must not be negative:" + hour);
        checkArgument(hour < 24,
                "hour must less than 24:" + hour);

        this.hour = hour % 12;
    }

    public Degree degree(LongHand longHand) {
        return Degree.of(360/12 * this.hour + additionalDegree(checkNotNull(longHand).minute));
    }

    private int additionalDegree(int minute) {
        return (minute == 0 ? 0 : 30/(60/minute));
    }
}

File: src/pooh3/mobi/app/a01_clock/model/ShortHand.kt
package pooh3.mobi.app.a01_clock.model

import pooh3.mobi.elseThrow

class ShortHand(hour: Int) {
    val hour: Int
    init {
        (hour >= 0).elseThrow("hour must not be negative:" + hour)
        (hour < 24).elseThrow("hour must less than 24:" + hour)
        this.hour = hour % 12
    }
}

fun ShortHand.degree(longHand: LongHand): Degree {
    return (360 / 12 * this.hour)
            .degree
            .plus(longHand.let {
                if (it.minute == 0)
                    0.degree
                else
                    (30 / (60 / it.minute)).degree
            })
}

val Int.shortHand: ShortHand
    get() = ShortHand(this)